ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -359,6 +357,116 @@ public void shouldNotEnforceTopicExistenceWhileParsing() {
     assertThat(parsedStatements.size(), equalTo(3));
   }
 
+  @Test
+  public void shouldThrowFromTryExecuteIfSourceTopicDoesNotExist() {
+    // Given:
+    final List<PreparedStatement<?>> statements = parse(
+        "CREATE STREAM S1 (COL1 BIGINT) "
+            + "WITH (KAFKA_TOPIC = 'i_do_not_exist', VALUE_FORMAT = 'JSON');");
+
+    // Expect:
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage(is("Kafka topic does not exist: i_do_not_exist"));
+
+    // When:
+    ksqlEngine.tryExecute(statements, KSQL_CONFIG, new HashMap<>());
+  }
+
+  @Test
+  public void shouldThrowFromExecuteIfSourceTopicDoesNotExist() {
+    // Given:
+    final List<PreparedStatement<?>> statements = parse(
```

Comment:
`shouldThrowFromTryExecuteIfSourceTopicDoesNotExist` and `shouldThrowFromExecuteIfSourceTopicDoesNotExist` differ only in one line. Maybe refactor the common code.

Reply:
Same for other tests, the shared code for tryExecute and execute can be refactored to make tests more concise.

Reply:
I'm happy with the code the way it is. Duplication in tests is less of an issue. What matters IMHO is readability. Moving common code into a function, e.g.

```java
  @Test
  public void shouldThrowFromTryExecuteIfSourceTopicDoesNotExist() {
    // Given:
    doCommonSetup();

    // When:
    ksqlEngine.tryExecute(statements, KSQL_CONFIG, new HashMap<>());
  }

  @Test
  public void shouldThrowFromExecuteIfSourceTopicDoesNotExist() {
    // Given:
    doCommonSetup();

    // When:
    ksqlEngine.execute(statements.get(0), KSQL_CONFIG, new HashMap<>());
  }
```

Makes the tests less readable, not more.